app/assets/javascripts/editor.js

```diff
@@ -104,16 +104,74 @@ class Editor {
   // h7() {
   //   this.wrap('#######','')
   // }
+
+  //debounce function addition
+  debounce(func, wait, immediate) {
+    var timeout;
+    return function () {
+      var context = this,
+        args = arguments;
+      var later = function () {
+        timeout = null;
+        if (!immediate) func.apply(context, args);
+      };
+      var callNow = immediate && !timeout;
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+      if (callNow) func.apply(context, args);
```

Comment:
I have less experience using JS than Rails, is there a reason for using var over the latest let here?

Reply:
`var` is used for global scope but I guess here it doesn't matter since the values it holds are accessed only by the debounce function. Thanks for the catch!!!